diff --git a/build.rs b/build.rs
index 8f88c02..7a77933 100644
--- a/build.rs
+++ b/build.rs
@@ -100,9 +100,9 @@ impl SwaggerZip {
 
         for index in 0..self.len() {
             let mut file = self.by_index(index)?;
-            let filepath = file
-                .enclosed_name()
-                .ok_or(ZipError::InvalidArchive("invalid path file"))?;
+            let filepath = file.enclosed_name().ok_or(ZipError::InvalidArchive(
+                std::borrow::Cow::Borrowed("invalid path file"),
+            ))?;
 
             if index == 0 {
                 zip_top_level_folder = filepath
@@ -281,7 +281,8 @@ fn download_file_reqwest(url: &str, path: PathBuf) -> Result<(), Box<dyn Error>>
         match parse_ca_file(&cainfo) {
             Ok(cert) => client_builder = client_builder.add_root_certificate(cert),
             Err(e) => println!(
-                "failed to load certificate from CARGO_HTTP_CAINFO `{cainfo}`, attempting to download without it. Error: {e:?}",
+                "failed to load certificate from CARGO_HTTP_CAINFO `{cainfo}`, attempting to \
+                 download without it. Error: {e:?}",
             ),
         }
     }
diff --git a/project.json b/project.json
new file mode 100644
index 0000000..4d9b40f
--- /dev/null
+++ b/project.json
@@ -0,0 +1,43 @@
+{
+  "name": "utoipa-swagger-ui",
+  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
+  "projectType": "library",
+  "sourceRoot": "target/patch/utoipa-swagger-ui-9.0.0/src",
+  "targets": {
+    "build": {
+      "executor": "@monodon/rust:check",
+      "outputs": [
+        "{options.target-dir}"
+      ],
+      "options": {
+        "target-dir": "dist/target/utoipa-swagger-ui"
+      }
+    },
+    "test": {
+      "cache": true,
+      "executor": "@monodon/rust:test",
+      "outputs": [
+        "{options.target-dir}"
+      ],
+      "options": {
+        "target-dir": "dist/target/utoipa-swagger-ui"
+      },
+      "configurations": {
+        "production": {
+          "release": true
+        }
+      }
+    },
+    "lint": {
+      "cache": true,
+      "executor": "@monodon/rust:lint",
+      "outputs": [
+        "{options.target-dir}"
+      ],
+      "options": {
+        "target-dir": "dist/target/utoipa-swagger-ui"
+      }
+    }
+  },
+  "tags": []
+}
